src/server/frontend_wayland/wl_surface.cpp

```diff
@@ -525,6 +528,17 @@ auto mf::NullWlSurfaceRole::scene_surface() const -> std::optional<std::shared_p
 {
     return std::nullopt;
 }
+
+void mf::WlSurface::set_fractional_scale(mir::frontend::FractionalScaleV1* fractional_scale)
+{
+    this->fractional_scale = wayland::Weak(fractional_scale);
```

Comment:
We would typically spell this
```suggestion
    this->fractional_scale = wayland::Weak{fractional_scale};
```